src/js/popup/api/gitlab.js

```diff
@@ -31,16 +43,26 @@ function getMergeRequestPath(projectId, mergeRequestId) {
 
 export function logWorkTime(hostname, token, projectId, mergeRequestId, spentTime) {
   const url = getApiRootUrl(hostname) + getMergeRequestPath(projectId, mergeRequestId) + `/add_spent_time?duration=${spentTime}s`;
-  return makeRequest('POST', url, token);
+  return makeRequest('POST', url, token).then((request) => request.response);
 }
 
 export function requestUserId(hostname, token) {
   const url = getApiRootUrl(hostname) + '/user';
-  return makeRequest('GET', url, token).then((response) => JSON.parse(response).id);
+  return makeRequest('GET', url, token).then((request) => JSON.parse(request.response).id);
 }
 
 export function requestNotes(hostname, token, projectId, mergeRequestId) {
-  const url = getApiRootUrl(hostname) + getMergeRequestPath(projectId, mergeRequestId) + '/notes?sort=asc';
-  return makeRequest('GET', url, token).then((response) => JSON.parse(response));
+  const onNotesLoad = (request) => {
+    const notes = JSON.parse(request.response);
+    const nextRequest = getNextRequest('GET', token, request.getResponseHeader('link'));
+    if (!nextRequest) return { notes };
+    return {
+      nextRequest: nextRequest.then(onNotesLoad),
+      notes,
+    };
+  };
+
+  const url = getApiRootUrl(hostname) + getMergeRequestPath(projectId, mergeRequestId) + '/notes?sort=asc&per_page=100';
```

Comment:
Template literals can be used here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals